### PR TITLE
Replace caf_io unit tests with Robot tests

### PR DESCRIPTION
**Pull request by Neverlord**
Get rid of the legacy I/O tests entirely. The unit tests that made sense to keep around were shifted ofter previously. Everything else got replaced with Robot tests.

### CODE REVIEW

**Comment by Neverlord** [comment]
@riemass we do take a small hit on the coverage here, but on the upside, it's >3k lines less code. The things that aren't covered are mostly things that we rip out for 1.0. Also, for some reason Coverity has no coverage for the Prometheus stuff even though we're testing it...

**Review by riemass — APPROVED** [review]
Looks good, approving it. There are two things that came to mind while reviewing the PR, if you don't mind sharing your opinion.

File: robot/CMakeLists.txt [hunk 1]
```diff
@@ -88,6 +87,22 @@ add_test(
...
+    CAF::core
+    CAF::io)
+
+add_test(
+  NAME "robot-middleman"
+  COMMAND
+    ${Python_EXECUTABLE}
+      -m robot
+      --variable BINARY_PATH:$<TARGET_FILE:caf-robot-middleman-driver>
+      "${CMAKE_CURRENT_SOURCE_DIR}/middleman/middleman.robot")
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply robot/CMakeLists.txt]: Good point. Fixed.

File: libcaf_io/caf/io/network/protocol.test.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+using namespace caf;
+using namespace std::literals;
+
+using caf::io::network::protocol;
+
+namespace {
+
+TEST("inspection") {
+  auto roundtrip = [this](auto x) {
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libcaf_io/caf/io/network/protocol.test.cpp:24]: Yeah, we use this in several tests and implement it anew each time. I'll do a followup for it.

**Pull request merged**